guppylang/tys/parsing.py

```diff
@@ -109,27 +107,81 @@ def arg_from_ast(
         # `ast.UnaryOp` negation of a `ast.Constant(5)`
         assert node.value >= 0
         nat_ty = NumericType(NumericType.Kind.Nat)
-        return ConstArg(ConstValue(nat_ty, node.value)), InputFlags.NoFlags
+        return ConstArg(ConstValue(nat_ty, node.value))
 
     # Finally, we also support delayed annotations in strings
     if isinstance(node, ast.Constant) and isinstance(node.value, str):
-        try:
-            [stmt] = ast.parse(node.value).body
-            if not isinstance(stmt, ast.Expr):
-                raise GuppyError("Invalid Guppy type", node)
-            set_location_from(stmt, loc=node)
-            shift_loc(
-                stmt,
-                delta_lineno=node.lineno - 1,  # -1 since lines start at 1
-                delta_col_offset=node.col_offset + 1,  # +1 to remove the `"`
-            )
-            return arg_from_ast(stmt.value, globals, param_var_mapping)
-        except (SyntaxError, ValueError):
-            raise GuppyError("Invalid Guppy type", node) from None
+        node = _parse_delayed_annotation(node.value, node)
+        return arg_from_ast(node, globals, param_var_mapping)
 
     raise GuppyError("Not a valid type argument", node)
 
 
+def _parse_delayed_annotation(ast_str: str, node: ast.Constant) -> ast.expr:
+    """Parses a delayed type annotation in a string."""
+    try:
+        [stmt] = ast.parse(ast_str).body
+        if not isinstance(stmt, ast.Expr):
+            raise GuppyError("Invalid Guppy type", node)
+        set_location_from(stmt, loc=node)
+        shift_loc(
+            stmt,
+            delta_lineno=node.lineno - 1,  # -1 since lines start at 1
+            delta_col_offset=node.col_offset + 1,  # +1 to remove the `"`
+        )
+    except (SyntaxError, ValueError):
+        raise GuppyError("Invalid Guppy type", node) from None
+    else:
+        return stmt.value
+
+
+def _parse_callable_type(
+    args: list[ast.expr],
+    loc: AstNode,
+    globals: Globals,
+    param_var_mapping: dict[str, Parameter] | None,
+) -> FunctionType:
+    """Helper function to parse a `Callable[[<arguments>], <return types>]` type."""
+    err = (
+        "Function types should be specified via "
+        "`Callable[[<arguments>], <return types>]`"
```

Comment:
Ditto
```suggestion
        "`Callable[[<arguments>], <return type>]`"
```
I wonder about also noting, this is standard python hyping!